### ISSUE TITLE
Implement fixed point bitsfx functions in llvm-libc

### ISSUE THREAD

**Issue by PiJoules** [llvm/llvm-project#113359]
Some fixed point functions from ISO 18037 are implemented in llvm-libc, but not all of them are implemented. The various bits`fx`  functions should also be added. Copying from the extension:

```
4.1.7.4 The bitwise fixed-point to integer conversion functions

The bitwise fixed-point to integer conversion functions bitsfx, where fx stands for one of hr, r, lr, hk, k, lk, uhr, ur, ulr, uhk, uk or ulk, take one fixed-point type argument (corresponding to fx); the type of the function is an implementation-defined integer type int_fx_t (for signed fixed-point types) or uint_fx_t (for unsigned fixed-point types), defined in the <stdfix.h> headerfile, that is large enough to hold all the bits in the fixed-point type.

The bitwise fixed-point to integer conversion functions return an integer value equal to the fixedpoint value of the argument multiplied by 2^F, where F is the number of fractional bits in the fixed point type. The result type is an integer type big enough to hold all valid result values for the given fixed-point argument type. For example, if the fract type has 15 fractional bits, then after the declaration

  fract a = 0.5;

the value of bitsr(a) is 0.5 * 2^15 = 0x4000.
```

These can first be implemented as `__builtin_*` functions in clang then llvm-libc can provide the wrappers for each of the builtin functions.

**Comment by braw-lee** [comment]
I can try this, assign me pls

**Comment by braw-lee** [comment]
also, are there any previously merged PR's that I can refer on how to implement libc functions?
@PiJoules

**Comment by lntue** [comment]
> also, are there any previously merged PR's that I can refer on how to implement libc functions? @PiJoules

Maybe something like https://github.com/llvm/llvm-project/pull/84391 (modulo the math part) to add new functions and tests to the `stdfix.h` header.

**Comment by PiJoules** [comment]
Initially I was thinking this could be a `__builtin_` function that would be exposed through llvm-libc but I think from https://github.com/llvm/llvm-project/issues/113357#issuecomment-2443183341, it would also work to just have this function do a `cpp::bit_cast` and avoid having to do clang-side changes.

**Comment by PiJoules** [comment]
@braw-lee if you're still working on this, you might be able to leverage some of the stuff added in https://github.com/llvm/llvm-project/pull/114912

**Comment by braw-lee** [comment]
oh thanks, that will be helpful

**Comment by krishna2803** [comment]
hi! @PiJoules @erichkeane  i would like to take this up

**Comment by krishna2803** [comment]
@PiJoules @erichkeane

**Comment by PiJoules** [comment]
> hi! [@PiJoules](https://github.com/PiJoules) [@erichkeane](https://github.com/erichkeane) i would like to take this up

Yup, thanks for taking this up! Sorry for the delayed response also
